axum/src/handler/into_service.rs

```diff
@@ -74,7 +74,8 @@ where
         use futures_util::future::FutureExt;
 
         let handler = self.handler.clone();
-        let future = Handler::call(handler, req).map(Ok::<_, Infallible> as _);
+        let future = Handler::call(handler, req);
+        let future = future.map(Ok as _);
```

Comment:
Tells Rust to try a bit harder to infer the type. Otherwise you get

```
error[E0308]: mismatched types
  --> axum/src/handler/into_service.rs:80:47
   |
80 |         super::future::IntoServiceFuture::new(future)
   |                                               ^^^^^^ expected fn pointer, found fn item
   |
   = note: expected struct `futures_util::future::Map<_, fn(Response<_>) -> Result<_, Infallible>>`
              found struct `futures_util::future::Map<_, fn(Response<_>) -> Result<_, _> {Result::<Response<UnsyncBoxBody<bytes::Bytes, axum_core::Error>>, _>::Ok}>`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `axum` due to previous error
```